src/write_prometheus.c

```diff
@@ -434,11 +434,17 @@ static void target_info(strbuf_t *buf,
       job = strdup(v);
       label_set_update(resource, "service.name", NULL);
     }
+    if (job == NULL) {
+      job = strdup(hostname_g);
+    }
 
     if ((v = label_set_get(*resource, "service.instance.id")) != NULL) {
       instance = strdup(v);
       label_set_update(resource, "service.instance.id", NULL);
     }
+    if (instance == NULL) {
+      instance = strdup("");
+    }
 
     strbuf_print(buf, "target_info{");
```

Comment:
Maybe also:
```
#ifdef EXPOSE_OPEN_METRICS
    strbuf_print(buf, "target{");
#else
    strbuf_print(buf, "target_info{");
#endif
```
(Or better yet, get the metric name to print from a variable added to earlier ifdef check.)

Reply:
According to https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#supporting-target-metadata-in-both-push-based-and-pull-based-systems `target_info` is correct. I'm not sure the doc is correct, but I'll hold off on changing this for now.

Reply:
I see, whereas prometheus metric is `target_info` of type `gauge`, OpenMetrics one is `target` metric of type `info`, hence the same name.